### PR TITLE
Onda_file_parsing

### PR DESCRIPTION
**Pull request by mofii**
A complete refactor of calibratrion loading using parse_onda()

The loading is now robust against different line endings (the motivator for this refactor)

The preamp gain (0 dB or 20 dB) and the calibration date is now specified in the calibration filename.

### CODE REVIEW

**Review by stevenaleung — CHANGES_REQUESTED** [review]
Nice refactor. Small change requested

File: src/onda_file_parsing.jl [hunk 1]
```diff
@@ -3,45 +3,124 @@
...
+
 """
-function parse_onda(name)
-    file = readlines(name)
+function parse_onda(filename)
+    data_in = read_file_as_strings(filename)
+    info, n_header_lines = parse_header(data_in)
+    info["DATA"] = parse_data(data_in, n_header_lines)
+    info["HEADER"] = get_raw_header(data_in)
+    info["device_name"] = create_device_name(info)
```
[the remark this answers is not in the document]
Comment (mofii) [reply src/onda_file_parsing.jl]: Yup. Done

File: src/onda_file_parsing.jl [hunk 2]
```diff
@@ -60,30 +64,37 @@ function found(field, line)
 end
 
 is_header_end(line) = occursin("HEADER_END", line)
-parse_data(data_in, nlines) = data_in[nlines + 1:end]
+function parse_data!(info)
+    nlines = info["n_header_lines"]
```
Comment (stevenaleung) [inline src/onda_file_parsing.jl:75]: Could refactor this to `nlines = get_num_header_lines(info)`

File: src/onda_file_parsing.jl [hunk 3]
```diff
@@ -8,25 +8,28 @@ Running the new file will load the calibration data to be used by HydrophoneCali
...
-    info["DATA"] = parse_data(data_in, n_header_lines)
-    info["HEADER"] = get_raw_header(data_in)
-    info["device_name"] = create_device_name(info)
-    write_to_file(info)
-    return nothing
+    info = Dict()
+    read_file_as_strings!(info, filename)
+    parse_header!(info)
+    parse_data!(info)
+    #data_in = info["data_in"] #, n_header_lines)
```
Comment (stevenaleung) [inline src/onda_file_parsing.jl:15]: Let's remove this commented line

**Review by stevenaleung — APPROVED** [review]
Looks good. Has room for a small future refactor

**Pull request merged**